packages/mobile/src/utils/insertAtBeginning.ts

```diff
@@ -1,4 +1,10 @@
 // inserts the new item at the first index and makes sure it's not present twice
 export const insertAtBeginning = <T>(item: T, items: T[]) => {
-  return [item, ...items.filter((anItem) => anItem !== item)]
+  return [
+    item,
+    ...items.filter(
+      (existingItem) =>
+        existingItem !== item && JSON.stringify(existingItem) !== JSON.stringify(item)
```

Comment:
This method works generically but seems a bit heavyweight. I wonder if there was a simpler `key` to use for the data we have?

Reply:
Good point, I'll find a better way to do this